weld/llvm.rs

```diff
@@ -765,223 +781,279 @@ impl LlvmGenerator {
         Ok(())
     }
 
+    /*********************************************************************************************
+    //
+    //
+    //
+    // TODO(shoumik): Refactor code above this point!
+    //
+    //
+    //
+    ********************************************************************************************/
+
+    /*********************************************************************************************
+    //
+    // Utilities
+    //
+    *********************************************************************************************/
+
+    /// Returns the Type, LLVM type (as a string), and LLVM symbol name (as a string).
+    fn symbol_info<'a>(&mut self, func: &'a SirFunction, sym: &Symbol) -> WeldResult<(&'a Type, String, String)> {
+        let ty = func.symbol_type(sym)?;
+        let llvm_ty = self.llvm_type(ty)?;
+        let llvm_str = llvm_symbol(sym);
+        Ok((ty, llvm_ty, llvm_str))
+    }
```

Comment:
btw the reason I made this change was that it took 3-4 lines of boilerplate code to get the name/type for *each* symbol, so it was making the code really long/messy.